--- lib/loader/router.ts
+++ lib/loader/router.ts
@@ -52,10 +52,13 @@
 }

 async function loadMiddlewares(base: string, middlewares: string[] = []) {
-  const midds = {}
+  const midds: { [key: string]: any[] } = {}
   for (const m of middlewares) {
     const middleware = await loadMiddleware(base, m)
-    Object.keys(middleware).map((name) => (midds[name] = [...(midds[name] || []), middleware[name]]))
+    for (const name in middleware) {
+      if (!midds[name]) midds[name] = []
+      midds[name].push(middleware[name])
+    }
   }
   return midds
 }
